### PR TITLE
DOCS: Added auto-generated docs for auto-generated code.

### PR DESCRIPTION
**Pull request by ekcoh**
### Description

Added auto-generated docs for auto-generated code based on ticket DOCF-6090 which contains user feedback about generated API surface not being documented. This is basically a bug.

### Testing status & QA

Tested during development by generating source code from Project-wide Actions asset.

### Overall Product Risks

Small, mainly affects code generator and related documentation.

- Complexity: Small
- Halo Effect: Small

### Comments to reviewers

Review for mistakes in generated documentation. Test by generating source code from various assets and verify that documentation is linked and looks correct via e.g. intellisense tooltips in VS and Rider.

### Checklist

Before review:

- [x] Changelog entry added.
    - Explains the change in `Changed`, `Fixed`, `Added` sections.
    - For API change contains an example snippet and/or migration example.
    - JIRA ticket linked, example ([case %<ID>%](https://issuetracker.unity3d.com/product/unity/issues/guid/<ID>)). If it is a private issue, just add the case ID without a link.
    - Jira port for the next release set as "Resolved".
- [ ] Tests added/changed, if applicable.
    - Functional tests `Area_CanDoX`, `Area_CanDoX_EvenIfYIsTheCase`, `Area_WhenIDoX_AndYHappens_ThisIsTheResult`.
    - Performance tests.
    - Integration tests.
- [ ] Docs for new/changed API's.
    - Xmldoc cross references are set correctly.
    - Added explanation how the API works.
    - Usage code examples added.
    - The manual is updated, if needed.

During merge:

- [ ] Commit message for squash-merge is prefixed with one of the list:
    - `NEW: ___`.
    - `FIX: ___`.
    - `DOCS: ___`.
    - `CHANGE: ___`.
    - `RELEASE: 1.1.0-preview.3`.

After merge:

- [ ] Create forward/backward port if needed. If you are blocked from creating a forward port now please add a task to ISX-1444.

### CODE REVIEW

**Review by jfreire-unity — APPROVED** [review]

**Review by duckets — APPROVED** [review]
Great to see this added, thank you!

File: Packages/com.unity.inputsystem/InputSystem/Editor/AssetImporter/InputActionCodeGenerator.cs [hunk 1]
```diff
@@ -41,6 +42,54 @@ public static class InputActionCodeGenerator
     {
         private const int kSpacesPerIndentLevel = 4;
 
+        private const string kClassExample = @"using namespace UnityEngine;
+using UnityEngine.InputSystem;
+
+// Example of using an InputActionMap named ""Player"" from a MonoBehavior implementing callback interface.
```
Comment (Pauliusd01) [inline Packages/com.unity.inputsystem/InputSystem/Editor/AssetImporter/InputActionCodeGenerator.cs]: Should be MonoBehaviour in the comment, super minor

**Review by Pauliusd01 — APPROVED** [review]
Read through the example and seems fine to me, generated code for PWA asset and a custom one and I see the changes and VS22 tooltips fine as well

File: Packages/com.unity.inputsystem/InputSystem/Editor/AssetImporter/InputActionCodeGenerator.cs (hunk 1 above)
Comment (ekcoh) [reply Packages/com.unity.inputsystem/InputSystem/Editor/AssetImporter/InputActionCodeGenerator.cs]: Thanks, its these kind of things I was sure I was going to mess up, will update

Comment (ekcoh) [reply Packages/com.unity.inputsystem/InputSystem/Editor/AssetImporter/InputActionCodeGenerator.cs]: Fixed in https://github.com/Unity-Technologies/InputSystem/pull/2045/commits/2658f2972f7da3741e6a020381f3d7daeac32e08

**Comment by ekcoh** [comment]
Unfortunately some post-review updates on this PR:
- https://github.com/Unity-Technologies/InputSystem/pull/2045/commits/032f3aa5469f95ead804253ff64168abf98577e0 fixes a formatting bug fix where a whitespace was inserted after doc-syntax prefix if text string was empty.
- https://github.com/Unity-Technologies/InputSystem/pull/2045/commits/a1b85852a81855d2a33ce33e0cd616be59710420 is a regeration of generated files in repo that didn't correctly reflect the changes to generator. These currently require manual regeneration if the original asset isn't reimported.
- https://github.com/Unity-Technologies/InputSystem/pull/2045/commits/aded9d0d26110126be70a9ae895417a19207dc42 was a formatting fix that shouldn't have been required if https://github.com/Unity-Technologies/InputSystem/pull/2045/commits/032f3aa5469f95ead804253ff64168abf98577e0 was committed first.

**Review by jfreire-unity — APPROVED** [review]

**Review by Pauliusd01 — APPROVED** [review]

**Pull request merged**